### PR TITLE
Convert createLocalTracks to TS

### PR DESCRIPTION
**Pull request by PikaJoyce**
Previous PR: https://github.com/twilio/twilio-video.js/pull/1687

This is the same as the previous PR with some reverts to the TS definitions and fixes to createLocalTracks to accommodate that.

**Contributing to Twilio**

> All third party contributors acknowledge that any contributions they provide will be made under the same open source license that the open source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### Before review
* [ ] Updated CHANGELOG.md if necessary
* [x] Added unit tests if necessary
* [ ] Updated affected documentation
* [x] Verified locally with `npm test`
* [x] Manually sanity tested running locally
* [ ] Included screenshot as PR comment (if needed)
* [ ] Ready for review

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
     "allowJs": true,
     "checkJs": false,
     "declaration": false,
+    "esModuleInterop": true,
```
Comment (makarandp0) [inline tsconfig.json:6]: @PikaJoyce, just curious - why was this needed?

**Review by charliesantos — APPROVED** [review]
